test/functional/feature_reindex_readonly.py

```diff
@@ -50,15 +51,33 @@ def reindex_readonly(self):
                     self.log.warning("Return early on Linux under root, because chattr failed.")
                     self.log.warning("This should only happen due to missing capabilities in a container.")
                     self.log.warning("Make sure to --cap-add LINUX_IMMUTABLE if you want to run this test.")
-                    return
-
-        self.log.debug("Attempt to restart and reindex the node with the unwritable block file")
-        with self.nodes[0].assert_debug_log(expected_msgs=['FlushStateToDisk', 'failed to open file'], unexpected_msgs=[]):
-            self.nodes[0].assert_start_raises_init_error(extra_args=['-reindex', '-fastprune'],
-                expected_msg="Error: A fatal internal error occurred, see debug.log for details")
+                    undo_immutable = False
```

Comment:
nit, might be good not to mix data types (probably a leftover from an earlier version of the PR?) :
```suggestion
                    undo_immutable = lambda: None
```
(also in the macOS/BSD branch below)

Reply:
Oops, indeed. In my mind `lambda: None` was falsy and and the if condition below wouldn't be executed, my bad.

Reply:
> It may be possible to type `lambda: pass`, which should be equal to `lambda: None`?

Unfortunately not:
```
$ python3
Python 3.10.11 (main, Apr 13 2023, 01:52:20) [Clang 13.0.0 ] on openbsd7
Type "help", "copyright", "credits" or "license" for more information.
>>> x = lambda: pass
  File "<stdin>", line 1
    x = lambda: pass
                ^^^^
SyntaxError: invalid syntax
```

I wonder what type checkers like mypy would consider as the correct falsy value for a lambda. Maybe `None` would be slightly better than `False`? (or maybe both would be rejected, idk). Anyways, `False` seems to do its job.